src/base/bittorrent/session.cpp

```diff
@@ -613,11 +613,14 @@ QString Session::tempPath() const
     return Utils::Fs::fromNativePath(m_tempPath);
 }
 
-QString Session::torrentTempPath(const InfoHash &hash) const
+QString Session::torrentTempPath(const TorrentInfo &torrentInfo) const
 {
-    return tempPath()
-            + static_cast<QString>(hash).left(7)
+    if (!torrentInfo.hasRootFolder() && (torrentInfo.filesCount() > 1))
```

Comment:
also minor, `torrentInfo.filesCount()` is cheaper to calculate, so maybe place it before `hasRootFolder()` check?